internal/testutil/reindent.go

```diff
@@ -10,19 +10,21 @@ import (
 func Reindent(in string) []byte {
 	var buf bytes.Buffer
 	var trim string
+	var trimSet bool
 	for _, line := range strings.Split(in, "\n") {
-		if trim == "" {
+		if !trimSet {
 			trimmed := strings.TrimLeft(line, "\t")
 			if trimmed == "" {
 				continue
 			}
-			if trimmed[0] == ' ' {
+			if len(trimmed) != len(line) && trimmed[0] == ' ' {
```

Comment:
A bit unclear how the extra test helps. In general that kind of belt-suspender makes sense when we're protecting from more expensive logic that we're about to do, but here the original comparison was already a single byte lookup being compared against a const, which shoud be almost as fast as it gets.

Reply:
This check is from the case where the entire string is space-indented. Without this check, reindenting `"    k: v"` will panic, complaining of mixed tabs and spaces despite there being no tabs.

On re-consideration though, I think this check is not useful for a different reason, which is that it will not cause a panic on invalidly _space_-indented strings (e.g `"    a\nb"`), which will be surprising as the function still panics on invalidly _tab_-indented ones (e.g `"\ta\nb"`).

I think the root problem here is the panic message being too specific, perhaps it should be weakened to "Space used in early indent in string: ..."?